HousingFinanceInterimApi/V1/Gateways/BatchLogErrorGateway.cs

```diff
@@ -20,14 +20,19 @@ public BatchLogErrorGateway(DatabaseContext context)
             _context = context;
         }
 
-        public async Task<BatchLogErrorDomain> CreateAsync(long batchId, string type, string message)
+        public async Task<BatchLogErrorDomain> CreateAsync(long? batchId, string type, string message)
         {
             try
             {
+                long concreteBatchId = batchId.HasValue ? batchId.Value : -1;
```

Comment:
BatchLogId is a foreign key to the BatchLog table, wouldn't this result in an error?

You could set it to e.g 1 or to a custom value in that table

Reply:
I haven't tested it yet.

I want the id to be very obviously unexpected value for whenever it's missing, so if possible I'll try sticking to -1.

I will give it a go at inserting a batchLog error record with -1 see what happens. Pretty much seeing if I can make it work. If I can't get it to work, I'll try seeing what's behind the id 1.

Reply:
Shoot, you're right it is a foreign key, and it does cause the error.

Was hoping for a quick fix. But these tables appear to be ,probably , among the only rare few for which the foreign keys actually configured within this db.

I tried adding 0 and -1 Id BatchLog entities, but Identity Create is disabled. Don't think it's worth re-enabling just for this.

I suppose I could just create a custom Id with a value of 10k+ and that would work, however, the secondary goal of this PR won't be fulfilled due to errors assigned to that batch log getting drowned out.

Will need to pause on this PR. And do some more thinking. Will mark it as Draft for the time being.